### PR TITLE
Implement task 5

### PR DESCRIPTION
**Pull request by f8B23Lef**
Add MovieDetails component

### CODE REVIEW

File: package-lock.json
Comment (khusanov-95) [inline package-lock.json:1]: this file shouldn't be it PR

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+    window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+  };
+
   return (
     <ErrorBoundary>
-      <Header />
-      <MainContent />
+      {showMovieDetails
+        ? (
+          <MovieDetails
```
Comment (khusanov-95) [inline src/App.js]: We can also put this components inside header and make conditional rendering there

File: src/components/MovieCard/MovieCard.js [hunk 2]
```diff
@@ -1,42 +1,75 @@
...
-    poster,
+    movie,
+    onEdit,
+    onDelete,
+    onClick,
   } = props;
 
-  const handleIconClick = () => {
-  };
+  const [hoverRef, isHovered] = useHover();
```
Comment (khusanov-95) [inline src/components/MovieCard/MovieCard.js:16]: Please use useRef hook for refs

File: src/components/MovieList/MovieList.js [hunk 3]
```diff
@@ -1,32 +1,73 @@
-import React from 'react';
-import poster from '../../assets/images/inceptionPoster.png';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import MovieCard from '../MovieCard/MovieCard';
+import { Modal, AddMovieModal, DeleteMovieModal } from '../../modals/index';
```
Comment (khusanov-95) [inline src/components/MovieList/MovieList.js]: Why AddMovieModal is used in multiple places ?

**Review by khusanov-95 — CHANGES_REQUESTED** [review]

File: src/components/MovieCard/MovieCard.js [hunk 4]
```diff
@@ -1,42 +1,75 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faEllipsisV } from '@fortawesome/free-solid-svg-icons';
+import ThreeDotsMenu from '../ThreeDotsMenu/ThreeDotsMenu';
+import { getYearFromDate } from '../../utils/utils';
+import { useHover } from '../../hooks/hooks';
```
Comment (khusanov-95) [inline src/components/MovieCard/MovieCard.js]: Why do we need this hook, we could just use onMouseEnter, onMouseLeave

File: src/components/MovieList/MovieList.js (hunk 3 above)
Comment (f8B23Lef) [reply src/components/MovieList/MovieList.js]: Fixed in prev PR. Created a separate EditMovieModal component.

File: src/components/MovieCard/MovieCard.js (hunk 4 above)
Comment (f8B23Lef) [reply src/components/MovieCard/MovieCard.js]: Just for reusability and training. I googled that this is one of the used ones and decided to redo this functionality into a hook.

File: package-lock.json
Comment (f8B23Lef) [reply package-lock.json:1]: Can we discuss this in a private chat?
There are different opinions about whether to commit or not `package-lock.json`.

File: src/components/MovieCard/MovieCard.js (hunk 2 above)
Comment (f8B23Lef) [reply src/components/MovieCard/MovieCard.js:16]: `useRef` is used inside `useHover`. Is this incorrect?
Do we need to use `useRef` in MovieCard and pass it as parameter to `useHover` similar to `useOutsideClick`?

File: src/App.js (hunk 1 above)
Comment (f8B23Lef) [reply src/App.js]: This was my first intention, but it seemed that it looked less readable and more overloaded.

Updated (moved MovieDetails inside Header).

**Review by khusanov-95 — APPROVED** [review]

**Pull request merged**